data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -384,6 +391,7 @@ private AccumulatingBulkRequest<BulkOperationWrapper, BulkRequest> createBulkReq
             final AccumulatingBulkRequest requestToReissue = bulkRequestSupplier.get();
             final ImmutableList.Builder<FailedBulkOperation> nonRetryableFailures = ImmutableList.builder();
             int index = 0;
+            List<BulkOperationWrapper> successfulOperations = new ArrayList<>();
```

Comment:
Initialize this array with the total number of operations, assuming the best case.